lib/Support/ResourceFinder.cpp

```diff
@@ -17,4 +21,20 @@ PathList ResourceFinder({
 
 });
 
+std::string getComponentsHash() {
```

Comment:
This function does not return an hash. Can we get a better name? `getComponentIDs`? `getComponentHashes`?

Reply:
> This function does not return an hash.

It returns the concatenation of all components' hashes, so in a certain way, it's an hash
Both name propositions have plurality, but in this case the return is a singular string which says `if this equals to the one saved, the versions are the same` (just as a "normal" hash)

Reply:
An hash certain properties that a string that's a concatenation of hashes does not have, so this is an hash if you consider the term in a very widened meaning, but OK.